animeal/src/Modules/Profile/ProfileViewController.swift

```diff
@@ -0,0 +1,212 @@
+import UIKit
+import UIComponents
+
+final class ProfileViewController: BaseViewController, ProfileViewModelOutput {
+
+    // MARK: - UI properties
+    private lazy var firstNameTextFieldView: TextFieldView = {
+        return TextFieldView(
+            model: TextFieldView.Model(
+                description: L10n.Profile.name,
+                placeholderText: L10n.Profile.name,
+                defaultText: viewModel.userFirstName,
+                textInputKind: .name
+            ),
+            delegate: self
+        )
+    }()
+
+    private lazy var lastNameTextFieldView: TextFieldView = {
+        return TextFieldView(
+            model: TextFieldView.Model(
+                description: L10n.Profile.surname,
+                placeholderText: L10n.Profile.surname,
+                defaultText: viewModel.userLastName,
+                textInputKind: .name
+            ),
+            delegate: self
+        )
+    }()
+
+    private lazy var emailTextFieldView: TextFieldView = {
+        return TextFieldView(
+            model: TextFieldView.Model(
+                description: L10n.Profile.email,
+                placeholderText: L10n.Profile.email,
+                defaultText: viewModel.userEmail,
+                textInputKind: .email
+            ),
+            delegate: self
+        )
+    }()
+
+    private lazy var phoneTextFieldView: TextFieldView = {
+        return TextFieldView(
+            model: TextFieldView.Model(
+                description: L10n.Profile.phoneNumber,
+                defaultText: viewModel.processedPhoneNumber,
+                isEnabled: false,
+                textInputKind: .phone
+            ),
+            delegate: self
+        )
+    }()
+
+    private lazy var calendarView: CalendarView = {
+        return CalendarView(
+            model: CalendarView.Model(
+                description: L10n.Profile.birthDate,
+                textFieldText: viewModel.formattedBirthdate
+            ),
+            datePickerView: datePickerView,
+            delegate: self
+        )
+    }()
+
+    private lazy var datePickerView: UIDatePicker = {
+        let datePicker = UIDatePicker()
+        datePicker.date = Date()
+        datePicker.locale = .current
+        datePicker.preferredDatePickerStyle = .inline
+        datePicker.datePickerMode = .date
+        datePicker.addTarget(self, action: #selector(onBirthdateDidChange), for: .valueChanged)
+        return datePicker
+    }()
+
+    private lazy var doneButton: UIButton = {
```

Comment:
It's button should be moved to ```UIComponents```, since it's could be reused in many places. However let's do it outside of this PR since this one already have a lot of code.

Reply:
Ok, I'll refactor this soon